source/package.template.html.erb

```diff
@@ -44,7 +44,7 @@
                       style="min-width: 225px"
                       id="version"
                       onchange="this.options[this.selectedIndex].value && (window.location = this.options[this.selectedIndex].value);">
-                <% for local_version, package_version in package.versions.sort.reverse %>
+                <% for local_version, package_version in package.versions.sort_by { |v| Gem::Version.new(v[0]) }.reverse %>
```

Comment:
Is `v` guaranteed to always have at least one element?

Or is there any case where v could be `nil` or empty?

It's been a while since my [poignant](https://en.wikipedia.org/wiki/Why%27s_(poignant)_Guide_to_Ruby) days, so I'll need a refresher:
- Is `v` = `local_version, package_version`, so `v[0]` = `local_version`?
- Therefore accessing `v[0]` is always safe?

Reply:
I feel like if it can be expanded to 2 variables, there must always be a 0 and a 1 indexed values.
And yes, from what I have tested, it looks like `v` = `local_version, package_version`, so `v[0]` = `local_version`

I am out of my Ruby depth here and when I got it working with `0` and saw that it was fine on all the pages I tested, I didn't look further :-)

`package.versions["<version_number>"]` is of the type `Middleman::Util::EnhancedHash` if it helps.
From what I understand the `sort`/`sort_by` function returns a sorted array, with the 0 element being the key and the 1 element being the value of the original Hash.

So, in the code above `v[0]` is the key for `package.versions["<version_number>"]`, e.g `<version_number>`, but `v[1]` returns all the attributes for `package.versions["<version_number>"]` (we actually don't use `package_version` in the code).

Reply:
tldr, yes, I think it is sage to use `v[0]`